### PR TITLE
Use new cmake paradigms

### PR DESCRIPTION
**Pull request by hjmjohnson**
## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,6 +45,8 @@ set(ITK_VERSION
     "${ITK_VERSION_MAJOR}.${ITK_VERSION_MINOR}.${ITK_VERSION_PATCH}")
 project(ITK
   VERSION ${ITK_VERSION}
+  DESCRIPTION "The Insight Toolkit (ITK) is an open-source, cross-platform toolkit for N-dimensional scientific image processing, segmentation, and registration."
+  HOMEPAGE_URL "https://github.com/InsightSoftwareConsortium/ITK"
```
Comment (N-Dekker) [inline CMakeLists.txt:49]: Shouldn't `HOMEPAGE_URL` be "https://itk.org"? Just a question, not a request  😃

**Review by dzenanz — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (hjmjohnson) [reply CMakeLists.txt:49]: I'm good with either URL. Originally I had https://itk.org but then put in the Github page.   Please let me know what the preference is.

Comment (N-Dekker) [reply CMakeLists.txt]: Thanks Hans. I'm good with either URL as well, I feel that it isn't really up to me anyway! Just want to be sure that you did consider https://itk.org, and apparently you did  😄

**Review by dzenanz — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,14 +12,8 @@
...
 set(ITK_NEWEST_VALIDATED_POLICIES_VERSION "3.19.7")
-cmake_minimum_required(VERSION ${ITK_OLDEST_VALIDATED_POLICIES_VERSION} FATAL_ERROR)
-if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "${ITK_NEWEST_VALIDATED_POLICIES_VERSION}")
-  #Set and use the newest available cmake policies that are validated to work
-  set(ITK_CMAKE_POLICY_VERSION "${CMAKE_VERSION}")
-else()
-  set(ITK_CMAKE_POLICY_VERSION "${ITK_NEWEST_VALIDATED_POLICIES_VERSION}")
-endif()
-cmake_policy(VERSION ${ITK_CMAKE_POLICY_VERSION})
+cmake_minimum_required(VERSION ${ITK_OLDEST_VALIDATED_POLICIES_VERSION}..${ITK_NEWEST_VALIDATED_POLICIES_VERSION} FATAL_ERROR)
```
Comment (issakomi) [inline CMakeLists.txt:15]: `cmake_minimum_required(VERSION <min>[...<policy_max>] [FATAL_ERROR])`

Looks like 3 dots?

Comment (hjmjohnson) [reply CMakeLists.txt]: Fixed.  Thanks for the close inspection.

**Pull request merged**